cmd/service-proxy/instrumentation.go

```diff
@@ -13,31 +13,25 @@ import (
 	"github.com/cloudoperators/greenhouse/pkg/common"
 )
 
-type ctxClusterKey struct {
-}
-type ctxNamespaceKey struct {
-}
-type ctxNameKey struct {
-}
-
 var (
 	clusterFromContext = promhttp.WithLabelFromCtx("cluster", func(ctx context.Context) string {
-		cluster, _ := ctx.Value(ctxClusterKey{}).(string) //nolint:errcheck
+		cluster, _ := ctx.Value(contextClusterKey{}).(string) //nolint:errcheck
 		return cluster
 	})
 
 	namespaceFromContext = promhttp.WithLabelFromCtx("namespace", func(ctx context.Context) string {
-		namespace, _ := ctx.Value(ctxNamespaceKey{}).(string) //nolint:errcheck
+		namespace, _ := ctx.Value(contextNamespaceKey{}).(string) //nolint:errcheck
 		return namespace
 	})
 
 	nameFromContext = promhttp.WithLabelFromCtx("name", func(ctx context.Context) string {
-		name, _ := ctx.Value(ctxNameKey{}).(string) //nolint:errcheck
+		name, _ := ctx.Value(contextNameKey{}).(string) //nolint:errcheck
 		return name
 	})
 )
 
-func InstrumentHandler(next http.Handler, registry prometheus.Registerer) http.Handler {
+func InstrumentHandler(pm *ProxyManager, registry prometheus.Registerer) http.Handler {
+	next := pm.ReverseProxy()
```

Comment:
Since this variable is only used within the call to `return injector(..)`. Could we call `pm.ReverseProxy()` directly?